apps/studio/src/components/sidebar/DetailViewSidebar.vue

```diff
@@ -157,18 +158,31 @@ export default Vue.extend({
     },
     processedValue() {
       const clonedValue = _.cloneDeep(this.value)
-      eachPaths(clonedValue, (path, value: string) => {
-        if (this.truncatedPaths.includes(path)) {
-          _.set(clonedValue, path, value.slice(0, globals.maxDetailViewTextLength))
+      eachPaths(clonedValue, (path, value) => {
+        if (this.jsonFields.includes(path)) {
+          _.set(clonedValue, path, JSON.parse(value as string))
+        } else if (this.truncatedPaths.includes(path)) {
+          _.set(clonedValue, path, (value as string).slice(0, globals.maxDetailViewTextLength))
         }
       })
       return clonedValue
     },
+    jsonFields() {
+      const fields = []
+      Object.keys(this.value).forEach((key) => {
+        const val = this.value[key]
+        if (typeof val === 'string' && getLanguageByContent(val).name === "json") {
```

Comment:
Is val ever already an object or array? If so `getLanguageByContent` would fail, as it expects a string.

Can we test this somehow?

Reply:
`val` can be a number or an ArrayBuffer, and it seems like `getLanguageByContent` only accepts strings.

Like a unit test for `getLanguageByContent`?

Reply:
Why aren't these values already objects when they're in the front-end? I'm super confused why we'd need to inflate them from strings.

Reply:
Ok, the problem isn't the front-end, it's the backend. It should be sending these back as objects.

Reply:
See for example `_text` fields are sent back correctly as arrays